Sources/GenericModule/View/ViewController.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Copyright © 2021 Rosberry. All rights reserved.
+//
+
+import UIKit
+
+open class ViewController<ViewModel: GenericModule.ViewModel, ViewInput, Output>: UIViewController, View {
+    public var output: Output
+    public var viewModel: ViewModel
+    private var viewOutput: ViewOutput
+
+    // MARK: - Lifecycle
+
+    public required init(viewModel: ViewModel, output: Output) {
+        self.viewModel = viewModel
+        self.output = output
+        guard let viewOutput = output as? ViewOutput else {
+            fatalError("`\(type(of: output))` does not conforms to `ViewOutput` protocol")
+        }
+        self.viewOutput = viewOutput
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+        viewOutput.viewDidLoad()
```

Comment:
I don't like that viewOutput viewDidLoad method now will be called before view controller configuration code (like adding subviews and etc) cause it may occur side effects.

Reply:
implemented special `viewDidLoad` with configuration handler that allows to call view configuration before presenter `viewDidLoad` (see example)